networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/gossip/topics/validation/AttesterSlashingValidatorTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.networking.eth2.gossip.topics.validation;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static tech.pegasys.teku.networking.eth2.gossip.topics.validation.InternalValidationResult.ACCEPT;
+import static tech.pegasys.teku.networking.eth2.gossip.topics.validation.InternalValidationResult.IGNORE;
+import static tech.pegasys.teku.networking.eth2.gossip.topics.validation.InternalValidationResult.REJECT;
+
+import com.google.common.eventbus.EventBus;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import tech.pegasys.teku.bls.BLSKeyPair;
+import tech.pegasys.teku.core.operationvalidators.AttesterSlashingStateTransitionValidator;
+import tech.pegasys.teku.datastructures.operations.AttesterSlashing;
+import tech.pegasys.teku.datastructures.util.DataStructureUtil;
+import tech.pegasys.teku.datastructures.util.MockStartValidatorKeyPairFactory;
+import tech.pegasys.teku.statetransition.BeaconChainUtil;
+import tech.pegasys.teku.storage.client.MemoryOnlyRecentChainData;
+import tech.pegasys.teku.storage.client.RecentChainData;
+
+public class AttesterSlashingValidatorTest {
+  private static final List<BLSKeyPair> VALIDATOR_KEYS =
+      new MockStartValidatorKeyPairFactory().generateKeyPairs(0, 25);
+  private DataStructureUtil dataStructureUtil = new DataStructureUtil();
+  private RecentChainData recentChainData;
+  private BeaconChainUtil beaconChainUtil;
+  private AttesterSlashingValidator attesterSlashingValidator;
+  private AttesterSlashingStateTransitionValidator stateTransitionValidator;
+
+  @BeforeEach
+  void beforeEach() {
+    recentChainData = MemoryOnlyRecentChainData.create(new EventBus());
```

Comment:
Isn't it preferable to use a real object? I thought we preferred using real objects over mocks.

Reply:
Well, I was going to suggest using `InMemoryStorageSystem.createEmptyV3StorageSystem` which is MORE real :). But then I thought since you only need the one method maybe mocking simpler.  And everything else is mocked basically.

Reply:
Switched to mocking recentChainData.